### ISSUE TITLE
Dummy may be connected to different server/map than main player

### ISSUE THREAD

**Issue by BannZay** [ddnet/ddnet#2804]
Something definitely gone wrong. What I did:
1. In browser I see Tropical Drag map
2. Press connect
3. I see "Tropical Drag" but client is crazy (check recording)
4. Server says that current map is Back in time

I connected with another client and could see my self just staying at the start of the "Back in time" map

Seems client guessing the map is "Tropical Drag" but in fact it is "Back in time".

Recording:
[2020-09-11 14-12-02.zip](https://github.com/ddnet/ddnet/files/5208556/2020-09-11.14-12-02.zip)

Connection log:
[connection_log.txt](https://github.com/ddnet/ddnet/files/5208568/connection_log.txt)

**Comment by heinrich5991** [comment]
From [Discord](https://discord.com/channels/252358080522747904/757720336274948198/1082685028368134206):

https://user-images.githubusercontent.com/366230/223470126-c27e0031-1daa-4e44-ab85-c57d2f98c9a0.mp4

**Comment by Robyt3** [comment]
Reported by Teero:

> if you are on a server that crashes with a map then restarts with a diffrent map and you then connect your dummy you will be on the restarted server on a diffrent map but your entities are wrong then

https://github.com/ddnet/ddnet/assets/23437060/1266923c-2f23-4cbb-ab85-094773673905

I can consistently reproduce the issue with this method.

**Comment by AssassinTee** [comment]
@Robyt3 this seems fixed to me be your PR?

**Comment by Robyt3** [comment]
No, it's still possible for the dummy to be connected to a different server (with a different map) than the main player.
